### DISCUSSION TITLE
The open-webui http://open-webui/api/v1/docs swagger document cannot open

### DISCUSSION THREAD

**Discussion by lewis-ing** [open-webui/open-webui#6942] [General]
The open-webui http://open-webui/api/v1/docs swagger document cannot open,
![image](https://github.com/user-attachments/assets/4812c811-a98e-4b3d-8618-bb7bf144dbb0)

so, install cmd: docker run -d -p 13000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui ghcr.io/open-webui/open-webui:main

**Comment by cgoer** [comment]
Had the same issue and had to define an env variable called "ENV" with the value "dev" for it to work.

docs are only displayed if ENV=dev, as found in `backend/open_webui/apps/webui/main.py:90`:
``` language=python
app = FastAPI(
    docs_url="/docs" if ENV == "dev" else None,
    openapi_url="/openapi.json" if ENV == "dev" else None,
    redoc_url=None,
)
```

Strangely, the env variable is set to dev as default in `backend/open_webui/env.py`, but that didn't work on my side either. Defining it fixed the problem for me. Hope it helps!

**Reply by JacksonSarah** [reply]
I am sorry to ask you that question, but where exactly in the env.py , I have to define the variable? I am quite a noob and kinda lost after trying for days :)

**Reply by cgoer** [reply]
My comment about the env.py-file is only a remark that "it should work even without doing anything" - but not a thing you need to set up there by yourself :)

You need to set the environment variable ENV=dev on startup to see the documentation. So e.g. if you run open-webui with docker, add the option `-e ENV=dev` to your `docker run`-command and it should work.

**Reply by JacksonSarah** [reply]
Thanks, for your explanation. 
That was indeed helpfull because I already tried that quite often and   thought my mistake, where somewhere here. You proved my ansible script to be not that wrong. Just a ""  to much around the value and it was finally solvable through 
that discussion
https://github.com/open-webui/open-webui/discussions/6978#discussioncomment-11835511 

somehow my adress is not http://my.ip.adress.yx:port/api/v1/docs
its http://my.ip.adress.xy:port/docs#

**Comment by kielpi** [comment]
Option -e ENV=dev for Docker works too, but it's not explicitly mentioned in the documentation.

**Comment by ausangshukla** [comment]
docker run -p 3000:8080 -e ENV=dev -v open-webui:/app/backend/data --name open-webui ghcr.io/open-webui/open-webui:main

Im still not able to see /docs 
How can I get the swagger api, and API docs does not seem to exist anywhere on the web either

TIA

**Reply by rgaricano** [reply]
I have it on  http://mainsite_or_ip/docs (or https if you have config ssl)
(and i didn't run dev option)

**Comment by stephenjoly** [comment]
Even though my docker compose looks like this

services:
openwebui:
container_name: openwebui
image: ghcr.io/open-webui/open-webui:main
restart: unless-stopped
environment:
- ENV="dev"
- WEBUI_URL="https://openwebui.mydomain.org/"
- ENABLE_OLLAMA_API=true
- OLLAMA_BASE_URL="http://192.168.1.23:11434/"
ports:
- "3007:8080"
volumes:
- $DOCKERDIR/appdata/open-webui:/app/backend/data

I still see 404 not found when navigating to https://openwebui.mydomain.org/api/v1/docs

I am using traefik for reverse proxy. Maybe that has something to do with it? I am trying to learn more about the api structure because im finding it imposisble to connect my ollama models to continue.dev via openwebui. I keep getting 403 errors in continue.dev.

**Reply by Kondratenko-Valeriy** [reply]
Kindly try [https://openwebui.mydomain.org/docs](https://openwebui.mydomain.org/docs) (without `/api/v1`)

**Reply by TuMee-Dev** [reply]
FWIW in my config /openapi.json worked but /docs did not. Traefix did need to be reconfigured.

 I had to add:
      - traefik.http.routers.open-webui-docs.rule=Host(`owui.${DOMAIN_NAME}`) && PathPrefix(`/docs`)
      - traefik.http.routers.open-webui-docs.service=open-webui
      - traefik.http.routers.open-webui-docs.tls=true
      - traefik.http.routers.open-webui-docs.entrypoints=websecure
      - traefik.http.routers.open-webui-docs.tls.certresolver=mytlschallenge
